pkg/policy/k8s/watcher.go

```diff
@@ -49,6 +55,124 @@ type policyWatcher struct {
 	// cidrGroupPolicies is the set of policies that contain ToServices references
 	toServicesPolicies map[resource.Key]struct{}
 	cnpByServiceID     map[k8s.ServiceID]map[resource.Key]struct{}
+	// maps cnp file name to cnp object. this is required to retrieve data during delete.
+	fileNameToCnpCache map[string]*types.SlimCNP
+}
+
+func (p *policyWatcher) translateToCNPObject(file string) (*cilium_v2.CiliumNetworkPolicy, error) {
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	jsonData, err := yaml.YAMLToJSON([]byte(data))
+	if err != nil {
+		return nil, err
+	}
+
+	cnp := &cilium_v2.CiliumNetworkPolicy{}
+	err = json.Unmarshal(jsonData, cnp)
+	if err != nil {
+		return nil, err
+	}
+
+	return cnp, nil
+}
+
+// read cilium network policy yaml file and convert to policy object and
+// add rules to policy engine.
+func (p *policyWatcher) addToPolicyEngine(cnpFileName string) error {
+	cnp, err := p.translateToCNPObject(cnpFileName)
+	if err != nil {
+		return err
+	}
+	slimCNP := &types.SlimCNP{
+		CiliumNetworkPolicy: cnp,
+	}
+	resourceKey := resource.NewKey(slimCNP)
+	resourceID := ipcacheTypes.NewResourceID(
+		ipcacheTypes.ResourceKindCNP,
+		slimCNP.ObjectMeta.Namespace,
+		slimCNP.ObjectMeta.Name,
+	)
+	err = p.onUpsert(slimCNP, resourceKey, k8sAPIGroupCiliumNetworkPolicyV2, resourceID)
```

Comment:
So the policy watcher logic here is made for K8s objects - the network policies get labels indicating that they were loaded from K8s, which is then displayed in Hubble for policy verdict events.

I'm a bit worried that this might cause user confusion - since all objects inserted here will have a Kubernetes name and namespace but do not exist in Kubernetes.

The policy repository itself however does have the concept of non-K8s policies. This is what I think we should rely on for loading policies from disk, rather than emulating K8s. 

By setting custom labels on the policy, we can then also indicate in Hubble and `cilium-dbg policy get` that those policies were indeed loaded from disk and perhaps even display the filename (or some other indication that they are not K8s policies but rather "super admin policies").

The policies are added to the repository here:
https://github.com/cilium/cilium/blob/57713adda56c2dfe3b6be68e0d13111381ebaf23/pkg/policy/k8s/cilium_network_policy.go#L140-L145

For policies loaded from file, I think we should use a different source (i.e. not `CustomResource`) and probably also set the rule labels to indicate their true origin (i.e. not have `io.cilium.k8s.policy.derived-from=CiliumNetworkPolicy`).

Reply:
In particular, the `onUpsert` function here also looks up K8s resources like `Service` and `CiliumCIDRGroup` objects, which can influence the set of allowed L3 targets of a policy. Given the "super admin" network policy motivation behind this PR, it might not be desirable for policies loaded from disk to refer to data from K8s.

I think we should consider moving the "load policy from disk" feature out of the K8s policy watcher and instead have a custom static file policy watcher that works independently from K8s.

Reply:
#1) The behavior should be same as that of how multiple conflicting network policies handled today and no need for special preference for these policies. Only difference is that this policy will not be shown as Kubernetes resource.
#2) Users need not be aware of these policies and these policies are mainly enforced by cloud administrator and user cannot configure this policy. The main usecase is for cloud admins to restrict user to access certain endpoints without blocking all.
#4) yes agreed. can add labels to these policy to differentiate this from k8s policy.

Just to point out, This PR is created based on what we discussed here: https://github.com/cilium/cilium/pull/30060#issuecomment-1955250995